lib/typeUtil.js

```diff
@@ -281,20 +281,35 @@ function clone(oldItem) {
   }
 }
 
+var VALID_ATTR_RE = /^[a-zA-Z][a-zA-Z0-9]*$/
+
 /**
  * @see http://docs.aws.amazon.com/amazondynamodb/latest/developerguide/ExpressionPlaceholders.html#ExpressionAttributeNames
  * @return {string} The alias. May just be the key itself if an alias is not needed.
  */
 function getAttributeAlias(key) {
-  return needsAttributeAlias(key) ? '#' + key : key
+  if (needsAttributeAlias(key)) {
+    // If this is just a reserved word, use # + word
+    if (VALID_ATTR_RE.test(key)) {
+      return '#' + key
+    }
```

Comment:
maybe split this into isReservedWord, isValidAttributeName, getReservedWordAlias, getInvalidNameAlias?